core/src/test/java/com/crawljax/core/state/StateFlowGraphTest.java

```diff
@@ -201,83 +196,46 @@ public void whenStateAddedTheGraphCouterIsIncremented() {
-		graph.putIfAbsent(state2);
-		graph.putIfAbsent(state3);
-		graph.putIfAbsent(state4);
-		graph.putIfAbsent(state5);
-
-		graph.addEdge(index, state2, newXpathEventable("/index/2"));
-		graph.addEdge(state2, index, newXpathEventable("/2/index"));
-		graph.addEdge(state2, state3, newXpathEventable("/2/3"));
-		graph.addEdge(index, state4, newXpathEventable("/index/4"));
-		graph.addEdge(state2, state5, newXpathEventable("/2/5"));
-		graph.addEdge(state4, index, newXpathEventable("/4/index"));
-		graph.addEdge(index, state5, newXpathEventable("/index/5"));
-		graph.addEdge(state4, state2, newXpathEventable("/4/2"));
-		graph.addEdge(state3, state5, newXpathEventable("/3/5"));
-
-		List<List<GraphPath<StateVertex, Eventable>>> results = graph.getAllPossiblePaths(index);
-
-		assertThat(results, hasSize(2));
-
-		assertThat(results.get(0), hasSize(5));
-
-		assertThat(results.get(0).get(0).getEdgeList(), hasSize(1));
-
-		assertThat(results.get(1), hasSize(2));
-		// int max = 0;
-		Set<Eventable> uEvents = new HashSet<>();
-
-		for (List<GraphPath<StateVertex, Eventable>> paths : results) {
-			for (GraphPath<StateVertex, Eventable> p : paths) {
-				// int z = 0;
-				for (Eventable edge : p.getEdgeList()) {
-					if (!uEvents.contains(edge)) {
-						uEvents.add(edge);
-					}
-
-				}
-			}
-		}
-	}
+    @Test
+    public void testAllPossiblePaths() {
+        graph.putIfAbsent(state2);
+        graph.putIfAbsent(state3);
+        graph.putIfAbsent(state4);
+        graph.putIfAbsent(state5);
+
+        graph.addEdge(index, state2, newXpathEventable("/index/2"));
+        graph.addEdge(state2, index, newXpathEventable("/2/index"));
+        graph.addEdge(state2, state3, newXpathEventable("/2/3"));
+        graph.addEdge(index, state4, newXpathEventable("/index/4"));
+        graph.addEdge(state2, state5, newXpathEventable("/2/5"));
+        graph.addEdge(state4, index, newXpathEventable("/4/index"));
+        graph.addEdge(index, state5, newXpathEventable("/index/5"));
+        graph.addEdge(state4, state2, newXpathEventable("/4/2"));
+        graph.addEdge(state3, state5, newXpathEventable("/3/5"));
+        graph.addEdge(state3, state4, newXpathEventable("/3/4"));
+
+        List<List<GraphPath<StateVertex, Eventable>>> results = graph.getAllPossiblePaths(index);
+
+        assertThat(results, hasSize(2));
+
+        assertThat(results.get(0), hasSize(5));
+
+        assertThat(results.get(0).get(0).getEdgeList(), hasSize(1));
+
+        assertThat(results.get(1), hasSize(2));
+
+        Set<Eventable> uEvents = new HashSet<>();
+        for (List<GraphPath<StateVertex, Eventable>> paths : results) {
+            for (GraphPath<StateVertex, Eventable> p : paths) {
+                for (Eventable edge : p.getEdgeList()) {
+                    if (!uEvents.contains(edge)) {
+                        uEvents.add(edge);
+                    }
+
+                }
+            }
+        }
+        assertThat(uEvents.size(), is(8));
```

Comment:
the test fails now. The reason is that 5 + 2 = 7 (not 8). I'll accept the PR once this is fixed.